pkg/proc/amd64_arch.go

```diff
@@ -200,11 +200,19 @@ func amd64SwitchStack(it *stackIterator, _ *op.DwarfRegisters) bool {
 
 		return true
 
-	case "runtime.goexit", "runtime.rt0_go", "runtime.mcall":
+	case "runtime.goexit", "runtime.rt0_go":
 		// Look for "top of stack" functions.
 		it.atend = true
 		return true
 
+	case "runtime.mcall":
+		if it.systemstack && it.g != nil && it.bi.GOOS == "freebsd" {
```

Comment:
This only happens on freebsd?

Reply:
I haven't seen anywhere else, the same tests don't fail on other OSes in CI. System calls are done differently on different OSes, I'm not surprised by this.

Reply:
Just curious if we've just gotten lucky on other OSes or if there's cause for implementing this elsewhere. If not, happy to merge this for now.